Several Gen 8 Move Effects

ghoulslash opened the pull request:
## Description
Adds:

1) Eerie Spell
2) Surging Strikes
3) Jungle Healing
![jungle_healing](https://user-images.githubusercontent.com/41651341/106195059-e453ee00-616c-11eb-86c0-9bcd926a922e.gif)
4) Triple Axel
5) Coaching
![coaching](https://user-images.githubusercontent.com/41651341/106195035-dd2ce000-616c-11eb-8092-a3724e179c5a.gif)
6) Lash out
7) Burning Jealousy
![burning_jealousy](https://user-images.githubusercontent.com/41651341/106194987-ce462d80-616c-11eb-9fc9-b7c97ee49d8a.gif)
8) Grassy Glide
9) Misty Explosion
10) Steel Roller (should probably merge #1295 first)
11) Obstruct
![obsruct](https://user-images.githubusercontent.com/41651341/106195080-eb7afc00-616c-11eb-8127-ff661764f96f.gif)
12) Behemoth Blade/Bash (sort of. sets up power boost with `#if B_DYNAMAX`)
13) Decorate
14) Snipe Shot

AmbientDinosaur commented:
This pull request needs to be tweaked a bit, with regard to Surging Strikes. As it is, Surging Strikes currently gets affected by the accuracy check that decides the amount of strikes Triple Kick (and Triple Axel) will do.

The multi-strike count of Surging Strikes is not affected by accuracy, it is always set to be three.

ultima-soul commented on src/battle_script_commands.c:
```diff
@@ -1171,13 +1171,15 @@ bool32 IsBattlerProtected(u8 battlerId, u16 move)
+        return TRUE;
     else if (gProtectStructs[battlerId].spikyShielded)
         return TRUE;
     else if (gProtectStructs[battlerId].kingsShielded && gBattleMoves[move].power != 0)
```
I wonder if some of these `gBattleMoves[move].power != 0` checks should be changed to `!IS_MOVE_STATUS(move)` checks since I think they're meant to represent a check for if it's a status move being used.

ultima-soul commented on src/battle_script_commands.c:
```diff
@@ -1252,7 +1254,7 @@ static bool32 TryAegiFormChange(void)
 {
     s32 i, moveType;
-
+    
```
```suggestion

```

ultima-soul commented on include/battle.h:
```diff
@@ -117,6 +117,7 @@ struct ProtectStruct
     u32 spikyShielded:1;
     u32 kingsShielded:1;
     u32 banefulBunkered:1;
+    u32 obstruct:1;
```
Maybe
```suggestion
    u32 obstructed:1;
```
to match the other protecting move flag names.

ultima-soul commented on src/battle_script_commands.c:
```diff
@@ -9016,11 +9137,16 @@ static u32 ChangeStatBuffs(s8 statValue, u32 statId, u32 flags, const u8 *BS_ptr
             index++;
             gBattleTextBuff2[index] = B_BUFF_EOS;
-
+            
```
```suggestion

```

ultima-soul replied:
```suggestion

```

ultima-soul commented on src/battle_util.c:
```diff
@@ -6686,6 +6688,20 @@ static u16 CalcMoveBasePower(u16 move, u8 battlerAtk, u8 battlerDef)
+        break;
+    case EFFECT_EXPLOSION:
+        if (move == MOVE_MISTY_EXPLOSION && gFieldStatuses & STATUS_FIELD_MISTY_TERRAIN && IsBattlerGrounded(battlerAtk))
+            basePower = 150;
```
This should probably be changed to a multiplication so that it's consistent with other damage boosts in this function.

ultima-soul reviewed (changes requested):
Some notes from testing:
```
Surging Strikes:
    - Should be like EFFECT_DOUBLE_HIT but for 3 hits since the 3 hits are guaranteed from a single accuracy calc

Obstruct:
    - Doesn't go down in chance after successive uses
    - Sucker Punch should say "But it failed!" when used on mon that used Obstruct

Behemoth Blade/Bash:
    - Maybe renamed `EFFECT_BEHEMOTH` to `EFFECT_DYNAMAX_DOUBLE_DMG`

Decorate:
    - Should be blocked by Crafty Shield

Snipe Shot:
    - Freezes the game when used
```

ghoulslash commented:
ready for re review

AsparagusEduardo reviewed (changes requested):
- Eerie Spell's PP drop message is bit too fast to read.
- I can't make Lash Out double its power
![lashout](https://user-images.githubusercontent.com/2904965/134265730-e622f5d7-db3e-4d24-85ec-835911f2ef7f.gif)
- Obstruct is lowering the defender's defense by 3 stages in addition to the attacker's 2 stages
![obstruct](https://user-images.githubusercontent.com/2904965/134265792-44ae365b-f9ba-417f-80af-03afe193c6ea.gif)
- Decorate is still not blocked by Crafty Shield.

ghoulslash commented:
ready for re-review:

`Eerie Spell's PP drop message is bit too fast to read.`
- added missing `waitmessage`

`I can't make Lash Out double its power`
- merged the statFell with Eject Pack so it should work now. One was set to ProtectStruct and the other SpecialStatuses

`Obstruct is lowering the defender's defense by 3 stages in addition to the attacker's 2 stages`
- fixed, merged with Buffel's `touchedProtectLike` implementation

`Decorate is still not blocked by Crafty Shield.`
- fixed

AsparagusEduardo commented:
Lash Out still doesn't want to double attack.
![lashout](https://user-images.githubusercontent.com/2904965/134418219-2afbf88c-368c-47aa-b1cc-c895aafcfd76.gif)
Everything else is ok.

AsparagusEduardo commented on src/battle_script_commands.c:
```diff
@@ -9409,7 +9409,7 @@ static u32 ChangeStatBuffs(s8 statValue, u32 statId, u32 flags, const u8 *BS_ptr
                 // Check eject pack. disableEjectPack set for edge cases (e.g. attacking weak armor'd eject pack holder with u-turn)
                 if (gProtectStructs[gActiveBattler].disableEjectPack == 0)
-                    gSpecialStatuses[gActiveBattler].statFell = TRUE;
+                    gProtectStructs[gActiveBattler].statFell = 1;
```
Why the change from `TRUE` to 1?
